src/rich-text/commands/index.ts

```diff
@@ -104,10 +105,38 @@ export function insertHorizontalRuleCommand(
         return false;
     }
 
-    dispatch &&
+    if (!dispatch) return true;
+
+    const docSize = Selection.atEnd(state.tr.doc).$anchor.pos;
+    const { $anchor, $head } = state.tr.selection;
+    const atEnd = docSize === Math.max($anchor.pos, $head.pos);
+
+    if (docSize === 1) {
```

Comment:
You may be interested in the [`insertParagraphIfAtDocEnd`](https://github.com/StackExchange/Stacks-Editor/blob/main/src/rich-text/commands/helpers.ts#L11) helper method here 😉

Reply:
Nice! I saw that at one point but completely forgot about it. Thanks for the nudge!

Reply:
This gets rid of the need for that `atEnd` condition but doesn't seem to do the trick for the empty editor condition. I honestly can't say why 🤷‍♂️

Reply:
It could be an issue with my implementation, but using `insertParagraphIfAtDocEnd` reintroduced the issue mentioned in https://github.com/StackExchange/Stacks-Editor/pull/124#issuecomment-1119265518. I've gone back to something close to what I had before using `insertParagraphIfAtDocEnd`.